### PR TITLE
fixing bug in name of Solender to avoid CLA troubles, is fixing …

### LINKED ISSUE #1117: Trigger unable to get subscriber uri across namespaces
**Describe the bug**
When a trigger/broker is in namespace A, it can not get the subscriber information for namespace B in order to forward events.

 **Expected behavior**
Trigger should be able to exist in namespace A and reference a service in namespace B

 **To Reproduce**
1) create namespaces "a" and "b"
2) create a broker in namespace "a"
3) create a service in namespace "b"
4) create a trigger in namespace "a" that references the broker in namespace a and references the subscriber in namespace "b"

 **Knative release version**
1.9.1

 **Additional context**
I already identified the location of the error. [Here](https://github.com/knative-sandbox/eventing-rabbitmq/blob/release-1.9/pkg/reconciler/trigger/trigger.go#L275) you will see that the subscriber ref namespace is overwritten to the namespace of the trigger without first checking if the namespace is empty. I am planning submit a pr to fix this myself, but wanted to document the issue for reference.

### PR DESCRIPTION
**Pull request by gabo1208**
# Changes

- :bug: Fix bug causing trigger subscriber ref namespace to be overwritten if provided

/kind bug

Fixes #1117

this PR is from @erictg to avoid getting this stale because of the CLA errors

### CODE REVIEW

**Comment by erictg** [comment]
@gabo1208 in the pr title, can you fix the spelling of my last name? It's "Solender" :)

**Comment by erictg** [comment]
Glad this is getting merged, I've been running a fork in prod for a few weeks now 😆

**Comment by gabo1208** [comment]
Yeah sometimes CLA can be a pain, but lets get this done :) with luck maybe today we'll have the approvals

**Comment by erictg** [comment]
Saw the codecov failed, those lines are tested by the unit tests I added. Not sure why codecov missed it. The 2 unit tests I added check the not defined path and different namespaces path.

File: pkg/reconciler/trigger/trigger_test.go [hunk 1]
```diff
@@ -449,6 +450,64 @@ func TestReconcile(t *testing.T) {
...
+						WithTriggerSubscribed(),
+						WithTriggerSubscriberResolvedSucceeded(),
+						WithTriggerStatusSubscriberURI(subscriberURI)),
+				}},
+			}, {
+				Name: fmt.Sprintf("%s: Creates everything with ref subscriber no ns provided", name),
+				Key:  testKey,
+				Objects: []runtime.Object{
+					ReadyBroker(config),
+					makeSubscriberAddressableAsUnstructured(),
```
Comment (kvmware) [inline pkg/reconciler/trigger/trigger_test.go:487]: This function https://github.com/knative-sandbox/eventing-rabbitmq/pull/1147/files#diff-c5ee2c23a8a6dd4c7aa3e6620d57f2bfd00a6b37dd072ade8bfd386dd4952d21L1406 populates a namespace. Therefor the codecoverage is not being covered in the line specified by codecov.

Comment (kvmware) [reply pkg/reconciler/trigger/trigger_test.go:487]: the function sets the name space to https://github.com/knative-sandbox/eventing-rabbitmq/pull/1147/files#diff-c5ee2c23a8a6dd4c7aa3e6620d57f2bfd00a6b37dd072ade8bfd386dd4952d21R78

**Comment by gabo1208** [comment]
The weird thing is that here it was passing @kvmware  https://github.com/knative-sandbox/eventing-rabbitmq/pull/1118

**Comment by kvmware** [comment]
/retest
KinD conformance tests / run (v1.25.9) was 18min stuck so cancelled and now re-running them

**Pull request merged**